### ISSUE TITLE
Null coalescing assignment: wrong type assumed 

### ISSUE THREAD

**Issue by j-schumann** [kalessil/phpinspectionsea#1877]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.7.1 |
| Language level |PHP 8.2                         |

Current behaviour
--
Code example
 ```php
$nc['groups'][] = "$type:read-single";
$nc['groups'][] = "$type:read-single-public";
$nc['groups'][] = "$type:read-single-private";

$nc[OpenApiFactory::OPENAPI_DEFINITION_NAME] ??= 'Read-Single';

if ($parentType) {
    $nc['groups'][] = "$parentType:read-single";
    $nc['groups'][] = "$parentType:read-single-public";
    $nc['groups'][] = "$parentType:read-single-private";
}
```
Shows wrong EA warnings: `[EA] Could provoke a PHP Fatal error ([] operator not supported for strings).`

<img width="573" alt="2023-01-28 12_03_46-FCP – AutoGroupResourceMetadataFactory php _ FCP - Multimandant_ 00_36_56" src="https://user-images.githubusercontent.com/114239/219396023-2ebebe59-5140-4d6b-b18e-0b934efaaa28.png">

When commenting L103 the [EA] errors disappear. Seems like the it assumes the whole array $nc is now a string after the assignment.
I don't know why this happens, I could not reproduce the behavior with a very simple demo:

```php
<?php
$a = [
    'l' => [],
];
$a['k'] ??= 'v';
$a['l'][] = 'w';
```
This *does not* show the same [EA] error

Expected behaviour
--
No warnings for nested arrays

Side Note
--

```php
<?php
$a = [
    'l' => [],
    'k' => 'm',
];
$a['k'] ??= 'v';
```

This hightlights the last line with `Condition is always 'true' because '$a' is evaluated at this point`, which is slightly incorrect as it is the the element inside $a that is evaluated. The inspection should read "...because '$a['k']' is evaluated..." or something similar, to make clear that the element is evaluated. When changing to `'k' => null` in L4 the inspection disappears correctly.

Environment details
---
PhpStorm PS-222.4459.28

**Comment by ea-inspections-team** [comment]
Thank you for reporting @j-schumann! This issue has already been addressed in #1938, so I am closing it as a duplicate.
